core/state/pruner/pruner.go

```diff
@@ -401,54 +404,121 @@ func dumpRawTrieDescendants(db ethdb.Database, root common.Hash, output *stateBl
 				output.Put(data.CodeHash, nil)
 			}
 			if data.Root != (common.Hash{}) {
-				// note: we are passing data.Root as stateRoot here, to skip the check for stateRoot existence in trie.newTrieReader,
-				// we already check that when opening state trie and reading the account node
-				trieID := trie.StorageTrieID(data.Root, key, data.Root)
-				storageTr, err := trie.NewStateTrie(trieID, sdb.TrieDB())
-				if err != nil {
-					return err
-				}
-				err = <-results
-				if err != nil {
-					return err
-				}
-				go func() {
-					threadsRunning.Add(1)
-					defer threadsRunning.Add(-1)
-					var err error
-					defer func() {
-						results <- err
-					}()
-					threadStartedAt := time.Now()
-					threadLastLog := time.Now()
-
-					storageIt, err := storageTr.NodeIterator(nil)
+				if !config.ParallelPrune {
```

Comment:
we are duplicating lots of code here,
can we abstract the storage traversal from start key to end key?

eg. we could have:
```
traverseStorage := func(startKey, endKey []byte) {
 ...
 // if endKey == nil then we skip endKey check
 ...
}
```

and then we could just:
```
if !config.ParallelPrune {
...
    go traverseStorage(nil, nil)
...
} else {
...
    for i := int64(1); i <= 32; i++ {
         ...
         if i == 32 {
             endKey = nil
         }
        go traverseStorage(startKey, endKey)
        ...
    }
...
}

```

Reply:
Makes sense, though instead of using keys as parameters, I have updated to use number of parts that we are dividing into 1 vs 32 as parameter to remove reductant code.

Let me know if this does not feel right.